ledger/complete/mtrie/flattener/decode.go

```diff
@@ -0,0 +1,147 @@
+package flattener
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/hash"
+)
+
+// DecodeNodeData reads the node data from reader, and uses scratch as reusable buffer for holding data read from disk.
+// it returns raw data so that they are allocated on stack instead of heap, useful when reading tons of nodes from a
+// checkpoint file.
+// any error returned means the file is malformated.
+func DecodeNodeData(reader io.Reader, scratch []byte) (
```

Comment:
This function is extracted from The encoding/ReadNode function, so that it can be reused by compare-checkpoint cmd